node/core/candidate-validation/src/lib.rs

```diff
@@ -421,6 +466,8 @@ async fn validate_candidate_exhaustive(
 			Ok(ValidationResult::Invalid(InvalidCandidate::ExecutionError(
 				"ambigious worker death".to_string(),
 			))),
+		Err(ValidationError::ArtifactNotFound) =>
+			Err(ValidationFailed("Code not found".to_string())),
```

Comment:
Nit: Let's try to make the error message a bit more clear. Imagine if you were going through the logs and just saw this error message. I imagine you wouldn't say thanks for the past you who left this message. If you were more descriptive, if it didn't light a light bulb, then it would perhaps at least help to grep the origin of the string.

```suggestion
			Err(ValidationFailed("validation host can't find artifact even though just received it".to_string())),
```

(I am not sure if you can apply this suggestion directly because of rustfmt and the line width constaints, and I am also not insisting in this particular wording)